test/common/ops/buy.js

```diff
@@ -1,14 +1,28 @@
 /* eslint-disable camelcase */
 
 import sinon from 'sinon'; // eslint-disable-line no-shadow
-
-let shared = require('../../common/script/index.js');
-
-describe('user.fns.buy', () => {
+import {
+  generateUser,
+} from '../../helpers/common.helper';
+import count from '../../../common/script/count';
+import buy from '../../../common/script/ops/buy';
+import predictableRandom from '../../../common/script/fns/predictableRandom';
+import randomVal from '../../../common/script/fns/randomVal';
+import content from '../../../common/script/content/index';
+import {
+  NotAuthorized,
+} from '../../../common/script/libs/errors';
+import i18n from '../../../common/script/i18n';
+
+describe('shared.ops.buy', () => {
   let user;
+  let fns = {
```

Comment:
When stubbing, this is not the same object that is being used in the source files. So it's not actually stubbing `predictableRandom` or `randomVal`. What I would do is create an `index.js` in the `fns` and `ops` folder that just export all the `fns` and `ops`, make those your entry point in the code and the tests. Then you can stub as needed.